src/test/regress/regress_gp.c

```diff
@@ -641,8 +643,24 @@ resGroupPalloc(PG_FUNCTION_ARGS)
 	if (!IsResGroupEnabled())
 		PG_RETURN_INT32(0);
 
+	if (startUpMBToAccount == -1)
+	{
+		startUpMBToAccount =
+			(VmemTracker_GetStartupChunks())
+			<< (VmemTracker_GetChunkSizeInBits() - BITS_IN_MB);
+	}
+
 	ResGroupGetMemInfo(&memLimit, &slotQuota, &sharedQuota);
 	size = ceilf(memLimit * ratio);
+	// At startup, the backend process is already consuming some amount of
+	// memory. In order not to complicate the logic of the tests, we take this
+	// memory into account when allocating memory for tests.
+	if (startUpMBToAccount)
```

Comment:
Why do we need this check?

Reply:
The purpose of this function is to allocate N% of the memory from the size of the current resource group. Because at the time the function is called, the backend is already consuming at least startup memory. Therefore, this function reduces memory allocation by the size of startup memory.